sentinel/src/transitions/accept_patient_reports.js

```diff
@@ -114,6 +114,38 @@ const addRegistrationToDoc = (doc, registrations) => {
   }
 };
 
+const addReportUUIDToRegistration = (doc, registrations) => {
+  if (registrations.length) {
+    var visitReportedDate = doc.reported_date;
+
+    var messageFound = false;
+    registrations.forEach(function(registration) {
+      if (registration.scheduled_tasks) {
+        registration.scheduled_tasks.forEach(function(task, index) {
+          if (!messageFound) {
+            if (
+              (task.state === 'delivered' || task.state === 'sent') &&
+              task.messages
+            ) {
+              var nextTask = registration.scheduled_tasks[index + 1];
+              if (
+                typeof nextTask !== 'undefined' &&
+                moment(nextTask.due) > moment(visitReportedDate)
+              ) {
+                registration.scheduled_tasks[index].report_uuid = doc._id;
+
+                db.medic.post(registration, function() {});
```

Comment:
This is not a good idea, here on as a general rule.

Think about the various edge cases that could occur by you throwing away the callback and not having this execute in line. Here are a couple I can think of, there are probably more:
 - If it errors for some reason we'll never know about it
 - When this transition has supposedly completed executing it actually hasn't: we can't prove that this write has taken place.

This needs to be refactored to use a callback or promise. Let me know if you want some guidance / more detail.